### PR TITLE
BUG: styler index and empty df

### LINKED ISSUE #43305: BUG: Styler IndexError when rendering an empty DataFrame
- [x] I have checked that this issue has not already been reported.

- [ ] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---
```python
df = pd.DataFrame()
midx = pd.MultiIndex.from_arrays([["A", "B"], [1, 2]], names=["zero", "one"])
df.style  # OK
df.index=[0,1]
df.style  # OK
df.index=midx
df.style  # IndexError
```

The error here was not the use of MultiIndex but of the index name display row when there are no columns.

### PR DESCRIPTION
**Pull request by attack68**
- [x] closes #43305 
- [x] tests added / passed
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/formats/style/test_style.py [hunk 1]
```diff
@@ -187,6 +187,12 @@ def test_render_trimming_mi():
     assert {"attributes": 'colspan="2"'}.items() <= ctx["head"][0][2].items()
 
 
+def test_render_empty_mi():
+    # GH 43305
+    df = DataFrame(index=MultiIndex.from_product([["A"], [0, 1]], names=[None, "one"]))
+    df.style.to_html()
```
Comment (alimcmaster1) [inline pandas/tests/io/formats/style/test_style.py:193]: Worth asserting this is as expected?

**Review by alimcmaster1 — CHANGES_REQUESTED** [review]
LGTM - Minor Comments

File: pandas/tests/io/formats/style/test_style.py (hunk 1 above)
Comment (attack68) [reply pandas/tests/io/formats/style/test_style.py:193]: not convinced. but I added it anyway. 👍

**Review by jreback — APPROVED** [review]
lgtm over to you @alimcmaster1

File: pandas/tests/io/formats/style/test_style.py (hunk 1 above)
Comment (alimcmaster1) [reply pandas/tests/io/formats/style/test_style.py]: Thanks - will save any regressions over time.

**Pull request merged**